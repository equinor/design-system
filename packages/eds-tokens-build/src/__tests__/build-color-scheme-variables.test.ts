import { describe, it, expect, beforeEach, afterEach } from 'vitest'
import { readFileSync, existsSync } from 'fs'
import path from 'path'
import {
  setupTestWorkspace,
  cleanupTestWorkspace,
  runScripts,
} from './test-utils'

describe('Color Scheme Variables Build', () => {
  let testWorkspace: string

  beforeEach(() => {
    testWorkspace = setupTestWorkspace()
  })

  afterEach(() => {
    cleanupTestWorkspace(testWorkspace)
  })

  it('should build CSS color scheme variables from tokens', () => {
    // Run full pipeline up to color scheme variables
    runScripts(
      [
        'generate-color-scheme-tokens.js',
        'generate-concept-tokens.js',
        'generate-semantic-tokens.js',
        'generate-dynamic-appearance-tokens.js',
        'build-color-scheme-variables.js',
      ],
      testWorkspace,
    )

    // Verify color scheme CSS was generated
    const colorSchemeCSS = path.join(
      testWorkspace,
      'build/css/color/color-scheme/color-scheme.css',
    )
    expect(existsSync(colorSchemeCSS)).toBe(true)

    // Snapshot test the generated CSS
    const actualColorScheme = readFileSync(colorSchemeCSS, 'utf-8')
    expect(actualColorScheme).toMatchSnapshot('color-scheme.css')
  })
})
