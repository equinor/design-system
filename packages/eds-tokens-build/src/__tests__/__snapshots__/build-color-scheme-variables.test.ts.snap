// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Color Scheme Variables Build > should build CSS color scheme variables from tokens > color-scheme.css 1`] = `
"/**
 * Do not edit directly, this file was auto-generated by scripts/build-color-scheme-variables.ts.
 */

:root,
[data-color-scheme="light"] {
  color-scheme: light;
}

[data-color-scheme="dark"] {
  color-scheme: dark;
}

:root {
  --x-color-bg-floating: light-dark(#feffff, #212222);
  --x-color-bg-backdrop: light-dark(#acaeb0, #818386);
  --x-color-bg-input: light-dark(#f5f5f6, #0b0b0b);
  --x-color-border-focus: light-dark(#74b9d7, #368eb1);
  --x-color-text-link: light-dark(#131f24, #e5ffff);
  --x-color-accent-1: light-dark(#ffede3, #0d0a0a);
  --x-color-accent-2: light-dark(#fffaf4, #291f1c);
  --x-color-accent-3: light-dark(#ffcab6, #8e4127);
  --x-color-accent-4: light-dark(#ffb89e, #ab431e);
  --x-color-accent-5: light-dark(#ff9f7d, #ce460c);
  --x-color-accent-6: light-dark(#ffc1aa, #8e4127);
  --x-color-accent-7: light-dark(#ff845b, #df4900);
  --x-color-accent-8: light-dark(#bf2600, #ff8558);
  --x-color-accent-9: light-dark(#b52300, #ffa685);
  --x-color-accent-10: light-dark(#952300, #ffc4ac);
  --x-color-accent-11: light-dark(#8a2300, #ffdbcb);
  --x-color-accent-12: light-dark(#a02200, #ffd2bf);
  --x-color-accent-13: light-dark(#2d160f, #fff6ee);
  --x-color-accent-14: light-dark(#ffcfbc, #472e26);
  --x-color-accent-15: light-dark(#ffffff, #040303);
  --x-color-neutral-1: light-dark(#f5f5f6, #0a0b0c);
  --x-color-neutral-2: light-dark(#feffff, #1d2325);
  --x-color-neutral-3: light-dark(#dadbdc, #376174);
  --x-color-neutral-4: light-dark(#cdcecf, #37718a);
  --x-color-neutral-5: light-dark(#bcbec0, #3585a4);
  --x-color-neutral-6: light-dark(#d3d4d5, #376174);
  --x-color-neutral-7: light-dark(#acaeb0, #368eb1);
  --x-color-neutral-8: light-dark(#67696b, #62bee5);
  --x-color-neutral-9: light-dark(#626366, #83d0f2);
  --x-color-neutral-10: light-dark(#515354, #a5e2fd);
  --x-color-neutral-11: light-dark(#4c4d4f, #c2f0ff);
  --x-color-neutral-12: light-dark(#57585a, #b7eaff);
  --x-color-neutral-13: light-dark(#1d1d1d, #e5ffff);
  --x-color-neutral-14: light-dark(#dddedf, #2b383d);
  --x-color-neutral-15: light-dark(#ffffff, #030303);
  --x-color-info-1: light-dark(#e8f8ff, #0a0b0c);
  --x-color-info-2: light-dark(#f5ffff, #1d2325);
  --x-color-info-3: light-dark(#c2e0ef, #376174);
  --x-color-info-4: light-dark(#add5e8, #37718a);
  --x-color-info-5: light-dark(#91c7df, #3585a4);
  --x-color-info-6: light-dark(#b8dbeb, #376174);
  --x-color-info-7: light-dark(#74b9d7, #368eb1);
  --x-color-info-8: light-dark(#077396, #62bee5);
  --x-color-info-9: light-dark(#086d8e, #83d0f2);
  --x-color-info-10: light-dark(#135a74, #a5e2fd);
  --x-color-info-11: light-dark(#16546c, #c2f0ff);
  --x-color-info-12: light-dark(#10617d, #b7eaff);
  --x-color-info-13: light-dark(#131f24, #e5ffff);
  --x-color-info-14: light-dark(#c7e3f1, #2b383d);
  --x-color-info-15: light-dark(#ffffff, #030303);
  --x-color-success-1: light-dark(#eff9e7, #0b0b0a);
  --x-color-success-2: light-dark(#fafff4, #20231d);
  --x-color-success-3: light-dark(#d0e2bf, #4d6335);
  --x-color-success-4: light-dark(#bfd7aa, #567435);
  --x-color-success-5: light-dark(#abca8e, #628835);
  --x-color-success-6: light-dark(#c8dcb5, #4d6335);
  --x-color-success-7: light-dark(#97bc71, #689236);
  --x-color-success-8: light-dark(#4e7711, #93c260);
  --x-color-success-9: light-dark(#4a7110, #aad47f);
  --x-color-success-10: light-dark(#3f5d15, #c2e5a2);
  --x-color-success-11: light-dark(#3b5717, #d7f3be);
  --x-color-success-12: light-dark(#426413, #cfedb3);
  --x-color-success-13: light-dark(#191f12, #f1ffe0);
  --x-color-success-14: light-dark(#d4e5c5, #31382a);
  --x-color-success-15: light-dark(#ffffff, #030303);
  --x-color-warning-1: light-dark(#ffede3, #0d0a0a);
  --x-color-warning-2: light-dark(#fffaf4, #291f1c);
  --x-color-warning-3: light-dark(#ffcab6, #8e4127);
  --x-color-warning-4: light-dark(#ffb89e, #ab431e);
  --x-color-warning-5: light-dark(#ff9f7d, #ce460c);
  --x-color-warning-6: light-dark(#ffc1aa, #8e4127);
  --x-color-warning-7: light-dark(#ff845b, #df4900);
  --x-color-warning-8: light-dark(#bf2600, #ff8558);
  --x-color-warning-9: light-dark(#b52300, #ffa685);
  --x-color-warning-10: light-dark(#952300, #ffc4ac);
  --x-color-warning-11: light-dark(#8a2300, #ffdbcb);
  --x-color-warning-12: light-dark(#a02200, #ffd2bf);
  --x-color-warning-13: light-dark(#2d160f, #fff6ee);
  --x-color-warning-14: light-dark(#ffcfbc, #472e26);
  --x-color-warning-15: light-dark(#ffffff, #040303);
  --x-color-danger-1: light-dark(#ffede9, #0d0a0a);
  --x-color-danger-2: light-dark(#fffaf6, #281f1e);
  --x-color-danger-3: light-dark(#ffcbc3, #89433b);
  --x-color-danger-4: light-dark(#ffb9af, #a4483e);
  --x-color-danger-5: light-dark(#fea194, #c64e42);
  --x-color-danger-6: light-dark(#ffc2b9, #89433b);
  --x-color-danger-7: light-dark(#fa897a, #d55145);
  --x-color-danger-8: light-dark(#b73129, #ff8272);
  --x-color-danger-9: light-dark(#ad2f27, #ffa496);
  --x-color-danger-10: light-dark(#8e2b23, #ffc2b8);
  --x-color-danger-11: light-dark(#842a23, #ffdad2);
  --x-color-danger-12: light-dark(#992c24, #ffd1c8);
  --x-color-danger-13: light-dark(#2b1714, #fff6f2);
  --x-color-danger-14: light-dark(#ffd0c8, #452f2c);
  --x-color-danger-15: light-dark(#ffffff, #040303);
}
"
`;
